collator/src/internal_queue/queue.rs

```diff
@@ -268,8 +286,35 @@ where
         Ok(())
     }
 
-    fn clear_session_state(&self) -> Result<()> {
-        self.diffs.clear();
-        self.session_state.truncate()
+    fn clear_uncommitted_state(&self) -> Result<()> {
```

Comment:
We call this method when block mismatched. Then we run sync to applied mc block and apply required queue diff for this mc block. Possibly we can get incorrect `diff_tail_len` because we keep `committed_diffs` unchanged. Should check this for all places were `clear_uncommitted_state` called.